extensions/yafw/src/index.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Form,
+  ActionPanel,
+  Action,
+  showToast,
+  Toast,
+  Clipboard,
+  useNavigation,
+  getSelectedFinderItems,
+  FileSystemItem,
+} from "@raycast/api";
+import { useEffect, useState } from "react";
+import { Videos } from "./videos";
+import { capitalizeSnakeCase, compressVideoFiles, fileName, normalizeFilePath, unique } from "./utils";
+import { fileExists } from "./utils";
+import { COMPRESSION_OPTIONS, CompressionOptionKey, DEFAULT_COMPRESSION, VIDEO_FORMATS } from "./constants";
+
+export default function Command() {
+  const [files, _setFiles] = useState<string[]>([]);
+  const [removedFiles, setRemovedFiles] = useState<string[]>([]);
+  const [clipboard, setClipboard] = useState<Clipboard.ReadContent | null>(null);
+  const [selectedFinderItems, setSelectedFinderItems] = useState<FileSystemItem[] | null>(null);
+  const [autoLoaded, setAutoLoaded] = useState(false);
+  const [compression, setCompression] = useState<CompressionOptionKey>(DEFAULT_COMPRESSION);
+  const navigation = useNavigation();
+
+  // Fake hook setter to ensure that perform some wanted operations
+  const setFiles = (newFiles: string[], automaticallyAdded = false) => {
+    newFiles = unique(newFiles.map(normalizeFilePath));
+
+    const isFileBeingRemoved = newFiles.length < files.length;
+    if (isFileBeingRemoved) {
+      const fileBeingRemoved = files.find((file) => !newFiles.includes(file));
+      setRemovedFiles([...removedFiles, fileBeingRemoved!]);
+    } else if (automaticallyAdded) {
+      setAutoLoaded(true);
+      // avoid adding files that we automatically added but the user removed
+      newFiles = newFiles.filter((file) => !removedFiles.includes(file));
+    } else {
+      // all new legit files should be deleted from the removed files list
+      setRemovedFiles(removedFiles.filter((file) => !newFiles.includes(file)));
+    }
+
+    newFiles = newFiles.filter(fileExists);
+
+    return _setFiles(newFiles);
+  };
+
+  useEffect(() => {
+    const lookForVideoSources = () => {
+      Clipboard.read().then(setClipboard);
+      getSelectedFinderItems()
+        .then(setSelectedFinderItems)
+        .catch(() => setSelectedFinderItems([]));
+    };
+    lookForVideoSources();
+    /* Poll for new files every 3 seconds
+     * In case the view is left in the background (i.e Raycast loses focus)
+     * the following warning will be shown at some point:
+     * > Looks like the command is rendering a lot without any changes. This might indicate an rendering loop caused by a `setState` not being wrapped in a `useEffect`.
+     * > This will degrade the performances of Raycast and Raycast might arbitrarily decide to terminate the extension if it happens too many times.
+     */
+    setInterval(lookForVideoSources, 3000);
```

Comment:
> Do you have a clue why the video I have selected seems to come back on every load (and on compleation)

this is probably caused by this interval, which I'm still not sure about. I made it so if you open the extension and only then you go to select/copy some videos, it regularly checks if there are more videos in the clipboard or the finder.

I could disable this behavior if you think is not a good idea

Reply:
I think it's the tradeoff that you'll need to keep it open, much better than using a lot of renders to keep it, maybe we can do something like this "warning"

![yafw 2024-04-25 at 08 45 25](https://github.com/raycast/extensions/assets/38033107/e94281d4-1a58-4803-af5f-53379a5edcba)

Reply:
hmmm it's not really during compression, it's right on that selection screen

user opens command -> no videos selected -> user goes to Finder to select or copy some videos -> comes back to the command -> the command auto-refreshes the list with the new files

to be honest I think I'm overthinking a bit here, I think I'll remove the setInterval and add a warning saying

> To add videos automatically, copy them or select them in the Finder and launch this command again

Reply:
☝🏻 this should also solve going back from the history and seeing the compressed video auto-selected

Reply:
<img width="48%" alt="image" src="https://github.com/raycast/extensions/assets/4324982/308d5fc1-9823-4e2e-aefb-9430a65e3d32">
<img width="48%" alt="image" src="https://github.com/raycast/extensions/assets/4324982/6835ec79-2aff-4f84-b5bc-b7802d71f9fb">